### ISSUE TITLE
Improve the doc blocks proptable with CSS-grid

### ISSUE THREAD

**Issue by ndelangen** [storybookjs/storybook#9635]
We have this code right here:
https://github.com/storybookjs/storybook/blob/992c98edd70b1bf536bf5ce69f796aad2d82bad5/lib/components/src/blocks/PropsTable/PropsTable.tsx#L11-L140

Which is used in the addon-docs to display the components metadata (propTypes) in a beautiful and useful way.

The component is currently a `<table>` with some styles.

But the widths and stylings of these elements is cumbersome and specifically the widths of the columns are not really what we want, and really hard to control.

We should change this component so it uses CSS-grid instead, so it becomes trivial to align columns correctly.

I asked for help on this issue on twitter: https://twitter.com/NorbertdeLangen/status/1220665994789736449

and I got replies from: @thefrontendwizard @codemacabre @maddesigns @alolalo
🙏 🙇 

Let's chat about the possible solutions, and divide up the work?

**Comment by ndelangen** [comment]
I'm happy to host a online meeting, if desired?

**Comment by ndelangen** [comment]
I've tried to create a date-selection poll here:
https://doodle.com/poll/5yhfimzek4uhfadz

If interested in a meeting about this, and also just learn a bit from each other, please cast a vote on when a meeting would be possible for you. 🙇 
@thefrontendwizard @codemacabre @maddesigns @alolalo

**Comment by codemacabre** [comment]
Thanks for sorting this @ndelangen, I've got a few things scheduled that I can potentially move around if my availability doesn't align with others'.

**Comment by ndelangen** [comment]
I have no idea what everyones timezone is, but here's a selection:
https://www.timeanddate.com/worldclock/meetingtime.html?iso=20200129&p1=16&p2=224&p3=179&p4=136

**Comment by eduina** [comment]
could help with the grid implementation if it's still needed

**Comment by codemacabre** [comment]
I've got a WIP fork of this (haven't had much time this week): https://github.com/codemacabre/storybook

I'm still getitng used to Emotion, but a rudimentary replacement of table (`<td>`, `<tr>`, etc.) elements with `<div>` elements and CSS grid appears to work fairly well.

**Comment by ndelangen** [comment]
I don't we'll want to use CSS subgrid then.

**Comment by ndelangen** [comment]
@codemacabre Need any help?

**Comment by codemacabre** [comment]
Apologies @ndelangen, I've had a busy few weeks and not made much progress recently. I've refactored the basic structure to use CSS Grid. A few styling updates are needed, then testing.

I'll get back on it this weekend, if you're happy for me to continue?

**Comment by ndelangen** [comment]
Yes, let me know if you need any help!

**Comment by codemacabre** [comment]
Hey @ndelangen apologies again for letting this get stale; not had any time to work on this recently.

However, it's worth rethinking whether `proptable` should actually be rewritten in CSS grid. Accessibility guidelines indicate that as `proptable` does actually contain tabular data, it would be better to keep it as a `<table>` element rather than using an admittedly better layout to mimic the appearance of tables.

Currently, the `proptable` can be navigated well with a screen reader and column headers are announced as expected (a major improvement would be to add headers at the beginning of each row, i.e. the 'name' cell should be `<th>` rather than a standard `<td>`). If this was re-written using CSS grid, the semantic meaning would be lost, even if it would make layout easier.

**Comment by ndelangen** [comment]
That's fine @codemacabre, thank you for your time & energy anyway!
